examples/contracts/aggregator_consumer/README.md

```diff
@@ -2,26 +2,179 @@
 
+## Prerequisites
 
 To get started, ensure that you have the following tools installed on your machine:
 
-- [starknet-foundry (v0.18.0)](https://github.com/foundry-rs/starknet-foundry/releases/tag/v0.18.0)
+- [starknet-foundry (v0.20.1)](https://github.com/foundry-rs/starknet-foundry/releases/tag/v0.18.0)
 - [scarb (v2.5.4)](https://github.com/software-mansion/scarb/releases/tag/v2.5.4)
 
-## Running Test Cases
+## Tests
 
-To run all test cases, you can use the following command:
+To run all test cases in the `tests/` directory, you can use the following command:
 
 ```sh
 snforge test
 ```
 
-## Running Scripts
+## Scripts
 
-To run a script, you can use the following command replacing the `NAME` argument with the name of the script to run:
+### Setup
 
-```sh
-make run-script NAME=example
-```
+#### Using a Local Starknet Docker Devnet
+
+If you would like to run the scripts against a local starknet devnet container:
+
+- First, execute the following command to run a [starknet-devnet-rs](https://github.com/0xSpaceShard/starknet-devnet-rs) container:
+
+  ```sh
+  make devnet
+  ```
+
+  If this command is re-run, it will fully stop the container and recreate it. This can be useful in case you'd like to restart from a completely clean state and re-run the deploy scripts.
+
+- Next, make sure that the `url` property in the `snfoundry.toml` file is configured such that it points to the docker container:
+
+  ```toml
+  [sncast.default]
+  url = "http://127.0.0.1:5050/rpc"
+  ```
+
+- The starknet devnet container comes with a set of prefunded accounts. In order to run the scripts, we'll need to add one of these accounts to our local accounts file (usually located at `~/.starknet_accounts/starknet_open_zeppelin_accounts.json`). This can be done using the following command:
+
+  ```sh
+  make add-account
+  ```
+
+- If you would like to view the account info later in the future, you can run:
+
+  ```sh
+  make view-accounts
+  ```
+
+At this point you should be ready to start executing scripts! Feel free to move onto the next section.
+
+#### Using Testnet
+
+If you would like to run the scripts against testnet:
+
+- First, make sure that the `url` property in the `snfoundry.toml` file is configured such that it points to Starknet testnet:
+
+  ```toml
+  [sncast.default]
+  url = "https://starknet-sepolia.public.blastapi.io/rpc/v0_7"
+  ```
+
+- Next, we'll need to create an account. This account will be used to pay for testnet transaction fees for some of the scripts. To generate an account (but not necessarily deploy it to the network yet), you can run the following command:
+
+  ```sh
+  make create-account
+  ```
+  
+- Once the account has been created, its info should be stored in an accounts file on your local machine (usually at `~/.starknet_accounts/starknet_open_zeppelin_accounts.json`). If you'd like to view the account information later on, you can use the following command:
```

Comment:
Let's define `accounts-file` to be in the current directory and gitignore it. That way we're not leaving global artifacts on user's system

Reply:
(only for devnet though)

Reply:
Done!

- In `snfoundry.toml`, made `accounts-file` configurable through an environment variable
- The `Makefile` is now the only place where the `accounts-file` path is defined - it is not hardcoded in any other place
- For devnet, the accounts file is called `accounts.json` and lives at the same level as `snfoundry.toml` it should be gitignored using the `.gitignore` file in the repo's root directory
- For testnet, the accounts file is unchanged (`~/.starknet_accounts/starknet_open_zeppelin_accounts.json`)